[fix] do not document '//TODO: ' comments

zsstrehli opened the pull request:
Fix for open issue:
https://github.com/realm/jazzy/issues/658

DangerCI commented:
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please include a CHANGELOG entry. <br />
You can find it at <a href="https://github.com/realm/jazzy/blob/master/CHANGELOG.md">CHANGELOG.md</a>.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Note, we hard-wrap at 80 chars and use 2 spaces after the last line.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

jpsim commented:
@zsstrehli this broke Objective-C doc generation. Please ping me again once you've fixed that and I'll be happy to review.

jpsim commented:
@zsstrehli thanks for continuing to work on this. Do you expect to be able to fix the broken tests any time soon?

zsstrehli commented:
@jpsim how can I retrigger the travis build? I don't see any build result related to my latest commit.

johnfairh commented on lib/jazzy/sourcekitten.rb:
```diff
@@ -339,7 +339,7 @@ def self.make_source_declarations(docs, parent = nil)
         declaration.type = SourceDeclaration::Type.new(doc['key.kind'])
         declaration.typename = doc['key.typename']
-        current_mark = SourceMark.new(doc['key.name']) if (declaration.type.mark? && (doc['key.name'].start_with?('MARK: ')))
+        current_mark = SourceMark.new(doc['key.name']) if (declaration.type.mark? && !(doc['key.name'].start_with?('TODO: ')))
```
To me, the earlier sourcekitten.rb patch looks a bit better than this one because it handles 'FIXME' as well as 'TODO'.

CI failed with the earlier patch because objc *#pragma mark something* lines come in to jazzy as type.marks as well, with their key.name being just the *something* part, no preceding 'mark'.

I think this patch misbehaves for pathological objc like `#pragma mark TODO: finish these methods` 😄 

The logic needed for current_mark seems to be "objc || (swift beginning 'MARK: ')".  A neat fix might be to change `SourceDeclaration::Type.mark?` to take a parameter of the key.name and do the test there -- the code here in `mark?` knows whether it is from an objc-pragma or not -- 2 calls to mark?() in sourcekitten.rb to update.

johnfairh commented:
hi @zsstrehli -- I think you should rebase your branch, the reason CI failed is because the branch predates jazzy moving from travis to circle!  The logic looks ok to me now; try running rubocop locally though, I think it will complain about line length at least (which will fail CI again).

zsstrehli commented:
@johnfairh Thanks for your help! I rebased my branch and also added a line break, but this time the tests could not be started: "bundler: failed to load command: bacon". Any idea?

orta commented:
@zsstrehli looks like a syntax error in sourcekitten.rb

```
SyntaxError: /Users/distiller/jazzy/lib/jazzy/sourcekitten.rb:643: syntax error, unexpected end-of-input, expecting keyword_end
```

will. see if I can spot it in the diff

orta commented on lib/jazzy/sourcekitten.rb:
```diff
@@ -359,7 +359,13 @@ def self.make_source_declarations(docs, parent = nil)
+          current_mark = SourceMark.new(doc['key.name']) 
+            if declaration.type.mark?
+        end
         if declaration.type.swift_enum_case?
```
there two `if`s may have once been on the same line as the `current_mark =`s?

jpsim commented:
Closing since this isn't reviewable as-is. Feel free to file a new PR.
